src/AuthorizationService.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         1.0.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Authorization;
+
+use Authorization\Policy\BeforePolicyInterface;
+use Authorization\Policy\Exception\MissingMethodException;
+use Authorization\Policy\ResolverInterface;
+
+class AuthorizationService implements AuthorizationServiceInterface
+{
+    /**
+     * Authorization policy resolver.
+     *
+     * @var \Authorization\Policy\ResolverInterface
+     */
+    protected $resolver;
+
+    /**
+     *
+     * @param \Authorization\Policy\ResolverInterface $resolver Authorization policy resolver.
+     */
+    public function __construct(ResolverInterface $resolver)
+    {
+        $this->resolver = $resolver;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function can(IdentityInterface $user, $action, $resource)
+    {
+        $policy = $this->resolver->getPolicy($resource);
+
+        if ($policy instanceof BeforePolicyInterface && !$policy->before($user, $resource)) {
+            return false;
+        }
+
+        $handler = $this->getHandler($policy, $action);
+
+        return $handler($user, $resource);
+    }
+
+    /**
+     * Returns a policy action handler.
+     *
+     * @param object $policy Policy object.
+     * @param string $action Action name.
+     * @return callable
+     * @throws \Authorization\Policy\Exception\MissingMethodException
+     */
+    protected function getHandler($policy, $action)
```

Comment:
Should this be abstracted? I have something like a `PolicyHandlerInterface` in mind. Implementations could call methods on policy objects in many way. This could be helpful if someone was porting their policies from another backend.

Reply:
I think until we have a use-case we shouldn't add extension points as they may not turn out to be that useful.

With that said, this protected method makes sense and keeps `can()` short and easy to read.